python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2

```diff
@@ -1474,13 +1474,14 @@ router bgp {{ router_bgp.as }}
 {%         if vrf.evpn_multicast is arista.avd.defined(true) %}
       evpn multicast
 {%             if vrf.evpn_multicast_gateway_dr_election.algorithm is arista.avd.defined %}
-{%                 if vrf.evpn_multicast_gateway_dr_election.algorithm == "preference" and vrf.evpn_multicast_gateway_dr_election.preference_value is arista.avd.defined %}
-{%                     set election_cli = "gateway dr election algorithm preference " ~ vrf.evpn_multicast_gateway_dr_election.preference_value %}
+{%                 if vrf.evpn_multicast_gateway_dr_election.algorithm == "preference" %}
+{%                     if vrf.evpn_multicast_gateway_dr_election.preference_value is arista.avd.defined %}
+{%                         set election_cli = "gateway dr election algorithm preference " ~ vrf.evpn_multicast_gateway_dr_election.preference_value %}
+{%                     endif %}
 {%                 else %}
 {%                     set election_cli = "gateway dr election algorithm " ~ vrf.evpn_multicast_gateway_dr_election.algorithm %}
 {%                 endif %}
          {{ election_cli }}
-         !
 {%             endif %}
```

Comment:
Sorry, but this will now fail with undefined variable in line 1484 in the case I mentioned.
```suggestion
{%             if vrf.evpn_multicast_gateway_dr_election.algorithm is arista.avd.defined %}
{%                 if vrf.evpn_multicast_gateway_dr_election.algorithm == "preference" %}
{%                     if vrf.evpn_multicast_gateway_dr_election.preference_value is arista.avd.defined %}
gateway dr election algorithm preference {{ vrf.evpn_multicast_gateway_dr_election.preference_value }}
{%                     endif %}
{%                 else %}
gateway dr election algorithm {{ vrf.evpn_multicast_gateway_dr_election.algorithm }}
{%                 endif %}
{%             endif %}
```